src/utils/utils.ts

```diff
@@ -148,3 +150,23 @@ export const calculateFileSize = (sizeInBytes: number): string => {
 
   return (sizeInBytes / Math.pow(THRESHOLD, exp)).toFixed(DECIMAL_PLACES) + SEPARATOR + units[exp];
 };
+
+interface ICheckColumnsCombinations {
+  columns: Array<string>;
+  combinations: Array<Array<string>>;
+}
+
+/**
+ * Function to check if the required combination has been met.
+ *
+ * @param columns - columns to be checked
+ * @param combinations - the required combination that need to exist together
+ *
+ */
+export const checkColumnsCombinations = ({ columns, combinations }: ICheckColumnsCombinations) => {
+  let isMismatch = false;
+  combinations.forEach((combination) => {
+    isMismatch = isMismatch || (combination.length >= 2 && combination.some((element) => !columns.includes(element)));
```

Comment:
You are going through all combinations even if you know for example the first one already failed.

Reply:
done